### PR TITLE
DI for sieves in processor. Simple test cases for processor and metri…

### PR DESCRIPTION
**Pull request by ppawelecsumo**
…ccache.

### CODE REVIEW

File: pkg/processor/metricfrequencyprocessor/processor_test.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+func createMetrics(metricsPerLibraryArgs ...map[string][]string) pdata.Metrics {
+	metrics := pdata.NewMetrics()
+	for _, metricsPerLibrary := range metricsPerLibraryArgs {
+		createRm(metricsPerLibrary).CopyTo(metrics.ResourceMetrics().AppendEmpty())
+	}
+
+	return metrics
+}
+
+func keySliceButDifferent(mapping map[string][]string) []string {
```
Comment (pmalek-sumo) [inline pkg/processor/metricfrequencyprocessor/processor_test.go:202]: The name is rather strange. Isn't it simple `getStringMapKeys`?

File: pkg/processor/metricfrequencyprocessor/processor_test.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+	}
+	resource2metrics := map[string][]string{
+		"lib-1": {"m1"},
+		"lib-3": {"m1", "m3"},
+		"lib-4": {"m2", "m3", "m4"},
+	}
+	input := createMetrics(resource1metrics, resource2metrics)
+	result, err := processor.ProcessMetrics(ctx, input)
+
+	assert.NoError(t, err)
```
Comment (pmalek-sumo) [inline pkg/processor/metricfrequencyprocessor/processor_test.go:74]:
```suggestion
	require.NoError(t, err)
```

File: pkg/processor/metricfrequencyprocessor/processor_test.go [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+	input := createMetrics()
+	result, err := processor.ProcessMetrics(ctx, input)
+
+	assert.NoError(t, err)
+	assert.Equal(t, result.ResourceMetrics().Len(), 0)
+}
+
+func TestKeepAllFromNonEmpty(t *testing.T) {
+	ctx := context.Background()
```
Comment (pmalek-sumo) [inline pkg/processor/metricfrequencyprocessor/processor_test.go:58]: nit: it's just used once, I guess no need to assign it to a variable, just use it in place 🤷

File: pkg/processor/metricfrequencyprocessor/processor_test.go [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+func TestKeepAllFromEmpty(t *testing.T) {
+	ctx := context.Background()
+	sieve := &keepAllSieve{}
+	processor := &metricsfrequencyprocessor{sieve: sieve}
+
+	input := createMetrics()
+	result, err := processor.ProcessMetrics(ctx, input)
+
+	assert.NoError(t, err)
```
Comment (pmalek-sumo) [inline pkg/processor/metricfrequencyprocessor/processor_test.go:53]:
```suggestion
	require.NoError(t, err)
```

**Review by pmalek-sumo — COMMENTED** [review]
I like this change 👍

File: pkg/processor/metricfrequencyprocessor/processor_test.go (hunk 4 above)
Comment (ppawelecsumo) [reply pkg/processor/metricfrequencyprocessor/processor_test.go]: switched in all relevant places

File: pkg/processor/metricfrequencyprocessor/processor_test.go (hunk 3 above)
Comment (ppawelecsumo) [reply pkg/processor/metricfrequencyprocessor/processor_test.go]: good idea, changed

File: pkg/processor/metricfrequencyprocessor/processor_test.go (hunk 1 above)
Comment (ppawelecsumo) [reply pkg/processor/metricfrequencyprocessor/processor_test.go]: I forgot to change it ^^ No overloading and no generics can sometimes be annoying. It's changed now.

**Review by pmalek-sumo — APPROVED** [review]

**Pull request merged**